java/client/src/main/java/glide/api/models/commands/scan/ScanOptions.java

```diff
@@ -0,0 +1,59 @@
+/** Copyright Valkey GLIDE Project Contributors - SPDX Identifier: Apache-2.0 */
+package glide.api.models.commands.scan;
+
+import java.util.ArrayList;
+import java.util.List;
+import lombok.experimental.SuperBuilder;
+
+/**
+ * This base class represents the common set of optional arguments for the SCAN family of commands.
+ * Concrete implementations of this class are tied to specific SCAN commands (SCAN, HSCAN, SSCAN,
+ * and ZSCAN).
+ */
+@SuperBuilder
+public abstract class ScanOptions {
```

Comment:
would you be able to rename to BaseScanOptions?  As per the description (that already calls it base class). 
We can then add ScanOptions for the Scan command itself.

Reply:
Sure. Did you want to do that here or in the ZScan PRs? Actually, do you want to put this in the same integration branch @acarbonetto